File: lib/js/js_boolean.js
'use strict';


function to_js_boolean(b) {
  if (b) {
    return true;
  }
  else {
    return false;
  }
}

var unsafe_bool_of_any = ( function(x){ return !x ? 0 : 1} );

function unsafe_boolean_of_any() {
  return ( !! x);
}

exports.to_js_boolean         = to_js_boolean;
exports.unsafe_bool_of_any    = unsafe_bool_of_any;
exports.unsafe_boolean_of_any = unsafe_boolean_of_any;
/* unsafe_bool_of_any Not a pure module */

File: jscomp/others/js_boolean.ml


type t = Js.boolean 

let to_js_boolean b = if b then Js.true_ else Js.false_

let unsafe_bool_of_any  =
    (Obj.magic [%bs.raw{| function(x){ return !x ? 0 : 1} [@arity 1]|}] :'a -> bool) 
    
let unsafe_boolean_of_any (type u) (x : u) :   t = 
    [%bs.raw{| !! x|}]
    